### PR TITLE
[Beta fix] Hides `GhostItemCardView` rendering behind feature flag

### PR DESCRIPTION
**Pull request by iamgabrielma**
## Description
This PR hides the `GhostItemCardView` behind a feature flag, which is currently not working appropriately in beta: The GhostItemCardView will not disappear when we reach the end of the product list. 

Up until now we've relied on the remote feature flag for POS features, but this won't be enough to hide the feature if we plan to open the feature to merchants soon, as they'll use 21.1 and see the work in progress.

This only hides the cosmetic part added by the `GhostItemCardView` on loading, infinite scrolling remains untouched as it's working correctly in beta.

Before
Simulator Screenshot - iPad mini (6th generation) - 2024-11-11 at 19 49 13

After

Simulator Screenshot - iPad mini (6th generation) - 2024-11-11 at 20 33 06

Question: Passing the feature flag from the app as we do right now, without reaching the ServiceLocator directly is a cross-cutting concern that we'll face more often now that we cannot rely just on the remote flag. Would if be simpler to use alternatives like `if debug` pre-compiler flags? or a feature flag service specifically for POS to keep it encapsulated within the "module"?

## Steps to reproduce
You can test this with a POS eligible store with just a couple of items, or by:
- Create a jurassic ninja site
- Create 1 simple product
- For simplicity, edit `POSEligibilityChecker.isEligible` to return `Just(true).eraseToAnyPublisher()`, so we do not need to comply with other POS requirements.
- Switch the scheme to a release build
- Confirm the ghostable cell does not appear when loading products.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/231">21.1 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by iamgabrielma** [comment]
cc: @rachelmcr as release manager, just for awareness 👍

File: WooCommerce/Classes/POS/ViewModels/ItemListViewModel.swift [hunk 1]
```diff
@@ -12,6 +12,8 @@ final class ItemListViewModel: ItemListViewModelProtocol {
 
     @Published private var currentPage: Int = Constants.initialPage
 
+    private(set) var shouldShowGhostableItemCard: Bool?
```
Comment (staskus) [inline WooCommerce/Classes/POS/ViewModels/ItemListViewModel.swift]: nitpick; no need to have boolean optional

```suggestion
    private(set) var shouldShowGhostableItemCard: Bool = false
```

File: WooCommerce/Classes/POS/Presentation/PointOfSaleEntryPointView.swift [hunk 2]
```diff
@@ -26,7 +26,9 @@ struct PointOfSaleEntryPointView: View {
                                               currencyFormatter: currencyFormatter,
                                               paymentState: .acceptingCard)
         let cartViewModel = CartViewModel(analytics: analytics)
-        let itemListViewModel = ItemListViewModel(itemProvider: itemProvider)
+
+        let shouldShowGhostableItemCard = ServiceLocator.featureFlagService.isFeatureFlagEnabled(.displayInfiniteScrollingUIDetailsInPointOfSale)
```
Comment (staskus) [inline WooCommerce/Classes/POS/Presentation/PointOfSaleEntryPointView.swift:30]: I agree with you. If we want to keep POS separate, maybe we can consider having a simple internal feature flag service for pos-specific feature flags.

**Review by staskus — APPROVED** [review]

File: WooCommerce/Classes/POS/ViewModels/ItemListViewModel.swift (hunk 1 above)
Comment (iamgabrielma) [reply WooCommerce/Classes/POS/ViewModels/ItemListViewModel.swift]: True! Updated on https://github.com/woocommerce/woocommerce-ios/pull/14373/commits/3b334d0aef5c8353a16ac1034a0225ff56e7e307

**Comment by rachelmcr** [comment]
The new [beta 21.1.0.1](https://github.com/woocommerce/woocommerce-ios/releases/tag/21.1.0.1) is released with this fix!

**Pull request merged**